cmd/oras/internal/option/packer.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package option
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/spf13/pflag"
+	"oras.land/oras-go/v2/content"
+)
+
+const (
+	annotationManifest = "$manifest"
+)
+
+var (
+	errAnnotationConflict    = errors.New("annotations cannot be specified via flags and file at the same time")
+	errAnnotationFormat      = errors.New("annotation MUST be a key-value pair")
+	errAnnotationDuplication = errors.New("annotation key duplication")
+)
+
+// Packer option struct.
+type Packer struct {
+	ManifestExportPath     string
+	PathValidationDisabled bool
+	AnnotationFilePath     string
+	ManifestAnnotations    []string
+
+	FileRefs []string
+}
+
+// ApplyFlags applies flags to a command flag set.
+func (opts *Packer) ApplyFlags(fs *pflag.FlagSet) {
+	fs.StringVarP(&opts.ManifestExportPath, "export-manifest", "", "", "export the pushed manifest")
+	fs.StringArrayVarP(&opts.ManifestAnnotations, "annotation", "a", nil, "manifest annotations")
+	fs.StringVarP(&opts.AnnotationFilePath, "annotation-file", "", "", "path of the annotation file")
+	fs.BoolVarP(&opts.PathValidationDisabled, "disable-path-validation", "", false, "skip path validation")
+}
+
+// ExportManifest saves the pushed manifest to a local file.
+func (opts *Packer) ExportManifest(ctx context.Context, fetcher content.Fetcher, desc ocispec.Descriptor) error {
+	if opts.ManifestExportPath == "" {
+		return nil
+	}
+	manifestBytes, err := content.FetchAll(ctx, fetcher, desc)
+	if err != nil {
+		return err
+	}
+	return os.WriteFile(opts.ManifestExportPath, manifestBytes, 0666)
+}
+
+// LoadManifestAnnotations loads the manifest annotation map.
+func (opts *Packer) LoadManifestAnnotations() (annotations map[string]map[string]string, err error) {
+	annotations = make(map[string]map[string]string)
```

Comment:
If there is no annotation, we should return a nil map instead of an empty map

Reply:
done